service/books_search/book_search_handler.go

```diff
@@ -0,0 +1,38 @@
+package books_search
+
+import (
+	"fmt"
+	"gin/config"
+	"gin/utils"
+	"github.com/gin-gonic/gin"
+	"net/http"
+)
+
+func validateSearchRequest(req *bookSearchRequest) error {
+	if req.Title == "" && req.AuthorName == "" && !(req.containsPriceFilter()) {
+		return fmt.Errorf("request must contain at least one filter")
```

Comment:
need "\n".
Also use `errors.New()` instead

Reply:
IMO this function should return a bool.

Reply:
why bool and not the error? Or do you mean both?

Reply:
I think this should handle the logic of `isValidSearchRequest`, and if it returns false then the calling function should return an error.
You don't have to, it's styling.